acquireTrial.m

```diff
@@ -79,10 +79,23 @@
 
 %% Process and scale data
 % code assumes that all modes of Multiclamp 700b have primary=membrane current & secondary=membrane potential
-data.current = rawData.Dev1_ai0 * rigSettings.current.softGain_pA; % convert to pA
-data.voltage = rawData.Dev1_ai1 * rigSettings.voltage.softGain_mV; % convert to mV
+% data.current = rawData.Dev1_ai0 * rigSettings.current.softGain_pA; % convert to pA
+% data.voltage = rawData.Dev1_ai1 * rigSettings.voltage.softGain_mV; % convert to mV
 
-% TODO write a funciton that loop over nidaq and save all fields into a sub
+% code detects primary and secondary outputs from MultiClamp commander
+[ampMeta] = telegraphCommander();
+
+primaryOutput = ampMeta.amp_primary_out_signal;
+
+if primaryOutput == 'Vm'
+    data.voltage = rawData.Dev1_ai0 * rigSettings.voltage.softGain_mV; % convert to mV
+    data.current = rawData.Dev1_ai1 * rigSettings.current.softGain_pA; % convert to pA
+else
+    data.current = rawData.Dev1_ai0 * rigSettings.current.softGain_pA; % convert to pA
+    data.voltage = rawData.Dev1_ai1 * rigSettings.voltage.softGain_mV; % convert to mV
```

Comment:
This rescaling is still using the values entered in ephysettings for softGain for voltage clamp or current clamp but it doesn't have to. Now that you have the meta data those gain values could change in Multiclamp and you could write the decoding to match that change.  
Basically I'm suggesting using values that are stored in ampMeta to calculate pA and mV correctly